src/Logger-HP_3456A/main.py

```diff
@@ -0,0 +1,196 @@
+        }
+
+    def set_GUIparameter(self):
+
+        GUIparameter = {
+            "Mode": list(self.modes.keys()),
+            "Resolution": list(self.resolutions.keys()),
+            "Range": list(self.ranges.keys()),
+            "NPLC": list(self.nplc_types.keys()),
+            "Auto-Zero": ["On", "Off"],
+            "Filter": ["Off", "On"],
+            "Display": ["On", "Off"],
+        }
+
+        return GUIparameter
+
+    def get_GUIparameter(self, parameter={}):
+
+        self.mode = parameter['Mode']
+        self.resolution = parameter['Resolution']
+        self.range = parameter['Range']
+        self.nplc = parameter['NPLC']
+        self.autozero = parameter['Auto-Zero']
+        self.filter = parameter['Filter']
+        self.display = parameter['Display']
+
+        self.port_string = parameter["Port"]
+
+        # here, the variables and units are defined, based on the selection of the user
+        # we have as many variables as channels are selected
+        self.variables = [self.mode]  # we add the channel name to each variable, e.g "Voltage DC"
+
+        self.units = [self.mode_units[self.mode]]
+
+        self.plottype = [True]  # True to plot data
+        self.savetype = [True]  # True to save data
+
+    def initialize(self):
+
+        self.port.write("CL1")  # reset all values
+
+    def deinitialize(self):
+
+        pass
+
+    def configure(self):
+
+        # Mode
+        self.port.write("%s" % self.modes[self.mode])
+
+        # Resolution
+        print(self.resolution)
+        print(self.mode)
+        if self.resolution in ["10 MOhm", "100 MOhm", "1 GOhm"] and self.mode in ["Voltage DC", "Voltage AC", "Voltage AC+DC"]:
+            msg = "Currently selected resolution is restricted to resistance measurement only"
+            raise Exception(msg)
+        else:
+            self.port.write("%s" % self.resolutions[self.resolution])
+
+        # Range
+        self.port.write("%s" % self.ranges[self.range])
+        
+        # NPLC Integration
+        self.port.write("%s" % self.nplc_types[self.nplc])
+        
+        # Auto-Zero
+        if self.autozero == "On":
+            self.port.write("Z1")
+        else:
+            self.port.write("Z0")
+        
+        # Filter
+        if self.filter == "On":
+            if self.mode in ["2W-Resistance", "4W-Resistance", "O.C. 2W-Res.", "O.C. 4W-Res."]:
+                msg = "Filter function cannot be used in Resistance mode"
+                raise Exception(msg)
+            else:
+                self.port.write("FL1")
+        else:
+            self.port.write("FL0")
+
+        # Display
+        if self.display == "Off":
+            self.port.write("D0")
+
+    def unconfigure(self):
+        if self.display == "Off":
+            self.port.write("D1")  # We switch Display on again if it was switched off
+
+    def measure(self):
+        self.data = self.port.read()  # retrieves current measurement data from the instrument
```

Comment:
I wonder how the measurement is triggered. Is the instrument permanently sending data? Could it be that there is old data in the buffer or is the readout somehow triggering a fresh measurement. Maybe adding a further comment?

Reply:
The standard setting uses the internal trigger which delivers the result to the output buffer. Once it is read, any following request has to wait for a new result. This is easily visible during normal use: raising the NPLC will increase the time between results proportionally as SweepMe has to wait for an answer until the integration is finished.

There is no specific measurement retrieval command, we just could use a software trigger instead of the internal trigger to conduct a measurement on the device. Retrieving it would be the same, however. But if you would rather promote having a manual trigger of the measurment for a timed start within e.g. the 'measure' procedure, I could implement it and try out.

Reply:
If there is a software trigger, I would recommend to use the software trigger during 'measure' as this starts the measurement and the readout of the result can be done during 'read_result' phase or also possible within 'measure' phase after the trigger. This ensures to have always a most recent meaurement in the buffer.
Using 'read_result' phase for readout has the advantage that you could use multiple multi-meters at the same time. They get all software triggered during 'measure' phase and the results can be read out one phase later.

A problem with the internal trigger can occur if you e.g. use multiple branches or a long delay and the buffer fills with data and the next time measure is used, an old data set is read out.

Sometimes, one needs to set the trigger back to internal after the measurement to see an update of the values on the display of the instrument again.

Reply:
Alright, I will adjust the driver next week and update my branch accordingly. No need to pull it prior to the changes.

Thanks for the hint regarding the switchback to the internal trigger, makes sense.

Reply:
New revision commited.